transcom/mymove
# INT B-18832 admin UI add edit page for requested office users (MERGE AFTER B-18831)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18832)

## Summary

Expanding upon [this PR](https://github.com/transcom/mymove/pull/12302) where we can see an overview page of the requested office user that was selected, this PR covers what happens when the edit button is clicked. This PR does contain code from B-18831 (link above), so please ensure that merges first.

This PR does the following:
- adds `RequestedOfficeUserEdit.jsx` component
- sprinkled some CSS in there
- removed incorrect `propType` in `AdminTimeout` to get rid of ugly red error in console
- added validation to check if `rejectionReason` is provided when rejecting, if not, throw an error on the screeen
- added validation to check if either `edipi` or `otherUniqueId` is provided, if not, throw an error on the screen

### How to test

1. Access MM as admin
2. Login and you should see the `Requested Office Users` screen
3. If empty, update some values in the `office_users` table to have a status of `REQUESTED`
4. Should see some folks now
5. Select one and click the `Edit` button (either one is fine)
6. Should now be seeing the screen in the screenshot below
7. Update some values, check some validations, and submit the form
8. You should see those values update in the db

## Screenshots
![Screenshot 2024-03-25 at 9 31 11 AM](https://github.com/transcom/mymove/assets/136510600/c43da9af-e1ee-4d3f-8469-9a54547a7dd7)
